docs/site/BelongsTo-relation.md

```diff
@@ -75,18 +98,44 @@ export interface OrderRelations {
 export type OrderWithRelations = Order & OrderRelations;
 ```
 
-The definition of the `belongsTo` relation is inferred by using the `@belongsTo`
-decorator. The decorator takes in a function resolving the target model class
-constructor and designates the relation type. It also calls `property()` to
-ensure that the decorated property is correctly defined.
+The above example can also be expanded with explicit relation and property
+definitions as follows,
+
+- The standard naming convention for the foreign key property in the source
+  model is `relation name` + `Id`. (e.g. Order.customerId).
+- If the property name in the source model has to be customized then the `name`
+  attribute in the relation definition (the relation name) has to be stated
+  explicitly. Also, the corresponding `hasMany` relation in the `target` model
+  has to be modified with the correct `keyTo` value.
+- In the following example :
+  - a relation definition is added to the decorator of the property with a
+    customized name (`cust_Id`).
+  - a property definition is added with the customized db column name
+    (`customer_id`).
+
+```ts
+class Order extends Entity {
+  // constructor, properties, etc.
+  @belongsTo(
+    () => Customer,
+    {keyFrom: 'customerId', keyTo: 'id', name: 'customer'},
+    {name: 'customer_id'},
```

Comment:
I think {`keyFrom`: '.., `name`: ..} is enough for customizing the name ( or even just to have the relation name {`name`: ..})

Reply:
this example is to provide all the parameters for the decorator

Reply:
@deepakrkris When looking at the [decorator's implementation](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/relations/belongs-to/belongs-to.decorator.ts#L23), I feel the 3rd parameter(property definition) is more for configuring the property, like:

- Is it required/generated/its lengh/default value, see [property configuration](https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html#general-property-properties), 
- And json schema config, see [type PropertyDefinition](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/model.ts#L29-L36)

`name` should be inferred from the property `cust_Id` itself.

Reply:
@jannyHou that is very strange, PropertyDefinition does not have a `name` key defined, how is typescript accepting the `name` attribute ? also I am using the explicit `name` key to customize db column name and it works perfectly fine !

Reply:
@jannyHou , also one more thing, `cust_Id` being a customized name (per standard naming convention it should be `customerId`), the `name` attribute cannot be inferred from the customized name, it has to be explicitly stated.

Reply:
@deepakrkris sorry for replying late (refreshed the PR several times last week but somehow git didn't show your message, weird...)

> PropertyDefinition does not have a name key defined, but typescript is accepting the name attribute

It's because type `PropertyDefinition` allows arbitrary properties, see [code](https://github.com/strongloop/loopback-next/blob/master/packages/repository/src/model.ts#L36)

> Also I am explicitly using the name key to customize db column names and it works perfectly fine for the todo-list example

Thank you got it, so it's used to configure the db column name.